modules/ui.py

```diff
@@ -429,18 +430,132 @@ def refresh(request: gradio.routes.Request):
     return refresh_button
 
 
-def create_upload_button(label, elem_id, destination_dir):
-    def upload(file):
+def create_upload_button(label, elem_id, destination_dir, model_tracking_csv="models.csv"):
+
+    model_list_csv_path = os.path.join(destination_dir, model_tracking_csv)
+
+    def verify_model_existence(hash_str):
+        if os.path.exists(model_list_csv_path):
+            with open(model_list_csv_path) as csvfile:
+                modelreader = csv.reader(csvfile, delimiter=',')
+                for file_hash_str, file_name in modelreader:
+                    if hash_str == file_hash_str:
+                        return file_name
+                return None
+        return None
+
+    def upload_file(file, hash_str):
         file_path = file.name
-        shutil.move(file_path, destination_dir)
-        return
-    upload_button = gr.UploadButton(label=label, elem_id=elem_id, file_types=[".ckpt", ".safetensors", ".bin"])
+        new_path = shutil.move(file_path, destination_dir)
+        with open(new_path,"rb") as f:
+            file_bytes = f.read() # read entire file as bytes

```

Comment:
This might be quite a few GB, which may cause issue? We may do it in the "chunk" way, or at somepoint.

Reply:
Yes. You absolutely right. We need to improve this in the next iteration.